plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/shipping-rates-control-package/package-rates.tsx

```diff
@@ -38,15 +38,9 @@ const PackageRates = ( {
 	const previousSelectedRateId = usePrevious( selectedRateId );
 
 	// Store selected rate ID in local state so shipping rates changes are shown in the UI instantly.
-	const [ selectedOption, setSelectedOption ] = useState( () => {
-		if ( selectedRateId ) {
-			return selectedRateId;
-		}
-		// Default to first rate if no rate is selected.
-		if ( rates.length > 0 ) {
-			return rates[ 0 ].rate_id;
-		}
-	} );
+	const [ selectedOption, setSelectedOption ] = useState(
+		selectedRateId ?? ''
```

Comment:
Empty initial state causes the mounting code on line 65 to select the initial rate. cc @senadir

Reply:
The logic above would select the selected one or the first one, why are we defaulting to `""`? this means having no selection. Do you mean the hook below would handle that?

Reply:
@senadir Look at the file as a whole and you'll see we have some on mount logic which selects the first method and ensures the on change event is fired. If we default to a rate here, no on change event is fired. So the data store would not be in sync with the component. Note local pickup uses `''` too for the same reason.